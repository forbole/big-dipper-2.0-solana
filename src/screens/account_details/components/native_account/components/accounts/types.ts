export type AddressesType = string[];

export type AccountsType = {
  stake: AddressesType;
  nonce: AddressesType;
  vote: AddressesType;
  token: AddressesType;
}
