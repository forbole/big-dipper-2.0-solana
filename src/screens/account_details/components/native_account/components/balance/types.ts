export type BalanceType = {
  native: TokenUnit;
  stake: TokenUnit;
  nonce: TokenUnit;
  vote: TokenUnit;
  total: TokenUnit;
}
