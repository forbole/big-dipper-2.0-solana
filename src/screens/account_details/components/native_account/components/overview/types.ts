export type OverviewType = {
  address: string;
}
