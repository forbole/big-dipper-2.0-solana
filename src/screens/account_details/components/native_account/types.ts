export type OverviewType = {
  address: string;
}

export type AccountType = {
  address: string;
}

export type AccountsType = {
  data: {
    [value:number]: AccountType[];
  };
  count: number;
  loading: boolean;
}

export type NativeAccountState = {
  loading: boolean;
  exists: boolean;
  overview: OverviewType;
}
