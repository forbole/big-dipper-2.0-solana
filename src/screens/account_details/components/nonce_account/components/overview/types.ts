export type OverviewType = {
  address: string;
  balance: number;
  authority: string;
  blockhash: string;
  fee: number;
}
