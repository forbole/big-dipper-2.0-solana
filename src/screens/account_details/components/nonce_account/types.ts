export type OverviewType = {
  address: string;
  authority: string;
  balance: TokenUnit;
  blockhash: string;
  fee: number;
}

export type NonceAccountState = {
  loading: boolean;
  overview: OverviewType;
}
