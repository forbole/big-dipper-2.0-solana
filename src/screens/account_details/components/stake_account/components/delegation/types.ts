export type DelegationType = {
  activated: TokenUnit;
  activating: TokenUnit;
  total: TokenUnit;
}
