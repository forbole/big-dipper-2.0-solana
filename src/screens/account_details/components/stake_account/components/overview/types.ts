export type OverviewType = {
  address: string;
  balance: number;
  stakeAuthority: string;
  withdrawAuthority: string;
  rentReserve: number;
  validator: string;
  delegated: number;
  rewards: number;
}
