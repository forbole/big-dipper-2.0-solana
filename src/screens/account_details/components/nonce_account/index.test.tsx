/* eslint-disable */
import React from 'react';
import { createMockClient } from 'mock-apollo-client';
import { ApolloProvider } from '@apollo/client';
import renderer from 'react-test-renderer';
import {
  MockTheme, wait,
} from '@tests/utils';
import { NonceAccountDetailsDocument } from '@graphql/types';
import NonceAccount from '.';

// ==================================
// mocks
// ==================================
jest.mock('next/router', () => ({
  useRouter: () => ({
    query: {
      address: "9XfsbcUJ8qM4esLtMocrMGLqRPkgYNbcmKLebtBzSjjZ",
    },
  }),
}));

jest.mock('./components', () => ({
  Overview: (props) => <div id="Overview" {...props} />,
}));

const mockData = jest.fn().mockResolvedValue({
  "data": {
    "nonceAccount": [
      {
        "address": "BcxTbVYxaZ9Uw92mDvQwWo8f8Kt4fnUGjaEvnyKvZFh",
        "authority": "38QU8LKVK1Ew5uzsqttamNTTFxvnfzgi2ACQvj3ekuom",
        "lamportsPerSignature": 5000,
        "blockhash": "CJm1NSWAm3NDDNtXZyCfK4oRukkS1kXKaWBn3CGQErHi",
        "nativeBalance": {
          "balance": 2500000
        }
      }
    ]
  }
});

// ==================================
// unit tests
// ==================================
describe('screen: NonceAccount', () => {
  it('matches snapshot', async () => {
    const mockClient = createMockClient();
    mockClient.setRequestHandler(
      NonceAccountDetailsDocument,
      mockData,
    );

    let component;
    renderer.act(() => {
      component = renderer.create(
        <ApolloProvider client={mockClient}>
          <MockTheme>
            <NonceAccount />
          </MockTheme>
        </ApolloProvider>,
      );
    });
    await wait();

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
