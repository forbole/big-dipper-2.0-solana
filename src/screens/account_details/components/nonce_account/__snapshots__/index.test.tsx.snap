// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: NonceAccount matches snapshot 1`] = `
<div
  className="makeStyles-overview"
  id="Overview"
  overview={
    Object {
      "address": "9XfsbcUJ8qM4esLtMocrMGLqRPkgYNbcmKLebtBzSjjZ",
      "authority": "38QU8LKVK1Ew5uzsqttamNTTFxvnfzgi2ACQvj3ekuom",
      "balance": Object {
        "baseDenom": "udaric",
        "displayDenom": "daric",
        "exponent": 6,
        "value": "2.5",
      },
      "blockhash": "CJm1NSWAm3NDDNtXZyCfK4oRukkS1kXKaWBn3CGQErHi",
      "fee": 5000,
    }
  }
/>
`;
